From e5b2559198fe900ef709fcc8a1a2936466ab7cd8 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 3 Jul 2024 11:48:22 +0200
Subject: [PATCH 0988/1695] drivers: stm32_tamp: rely on DT for initial tamper
 pin secure state

Remove call to stm32_gpio_set_secure_cfg() that is not required as the
tamper feature prevails on the GPIO. For pin secure level at platform
init, we should rely on the device tree.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a12db4b6b54807b0deaf6a6100fb41f715336d46 drivers: stm32_tamp: rely on DT for initial tamper pin secure state]
Change-Id: I54af98d0cba48a1fbde86385d09ff6aecb3aeee7
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391135
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_tamp.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/core/drivers/stm32_tamp.c b/core/drivers/stm32_tamp.c
index 1ff83ac55..d214f7362 100644
--- a/core/drivers/stm32_tamp.c
+++ b/core/drivers/stm32_tamp.c
@@ -1548,9 +1548,6 @@ static void stm32_tamp_configure_pin(uint32_t id, struct gpio *gpio, bool out,
 			break;
 		}
 	}
-
-	/* TAMPER pins are always secure (without effect, but keep coherency) */
-	stm32_gpio_set_secure_cfg(bank, pin, true);
 }
 
 static TEE_Result
-- 
2.47.2

